### PR TITLE
Fix storage-tests again

### PR DESCRIPTION
**Pull request by jasagredo**
# Description

Please include a meaningful description of the PR and link the relevant issues
this PR might resolve.

Also note that:

- New code should be properly tested (even if it does not add new features).
- The fix for a regression should include a test that reproduces said regression.

### CODE REVIEW

**Comment by jasagredo** [comment]
* **#1002** <a href="https://app.graphite.dev/github/pr/IntersectMBO/ouroboros-consensus/1002?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1005** <a href="https://app.graphite.dev/github/pr/IntersectMBO/ouroboros-consensus/1005?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `utxo-hd-8.9`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jasagredo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/Impl/Snapshots.hs [hunk 1]
```diff
@@ -181,18 +182,16 @@ writeExtLedgerState (SomeHasFS hasFS) encLedger path cs = do
...
-  -> SomeHasFS m
-  -> SomeHasFS m
+  -> [SomeHasFS m]
   -> SnapshotPolicy
   -> m [DiskSnapshot]
-trimSnapshots tracer hasFS ssdhasFS SnapshotPolicy{onDiskNumSnapshots} = do
+trimSnapshots tracer fss SnapshotPolicy{onDiskNumSnapshots} = do
     -- We only trim temporary snapshots
-    diskSnapshots <- filter diskSnapshotIsTemporary <$> listSnapshots hasFS
+    diskSnapshots <- filter diskSnapshotIsTemporary . concat <$> mapM listSnapshots fss
```
Comment (jasagredo) [inline ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/Impl/Snapshots.hs:190]: This is wrong, I need to find a better way to do this. We won't drop enough snapshots as they will be doubled...

File: ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/LedgerDB/StateMachine.hs [hunk 2]
```diff
@@ -460,40 +465,35 @@ openLedgerDB flavArgs env cfg fs = do
 data Environment =
   Environment
     (LedgerDB' IO TestBlock)
-    (OnDemandActions' IO TestBlock)
```
Comment (jorisdral) [inline ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/LedgerDB/StateMachine.hs:463]: If you want to get rid of the `undefined`s, you could use `MVar`s here instead, which would initially be empty

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/V1/Init.hs [hunk 3]
```diff
@@ -300,25 +300,43 @@ mkInternals ::
...
-    , truncateSnapshot = \ss -> getEnv h $ \env -> do
-        truncateFile (ldbStateHasFS env) (snapshotToStatePath ss)
-        truncateFile (ldbTablesHasFS env) (snapshotToTablesPath ss)
-    , flushNOW = getEnv h $ \env ->
-        withWriteLock
-          (ldbLock env)
-          (flushLedgerDB (ldbChangelog env) (ldbBackingStore env))
+    , wipeLedgerDB = getEnv h $ \env ->
+        mapM_ destroySnapshots [ldbTablesHasFS env, ldbStateHasFS env]
     , closeLedgerDB = getEnv h $ \env -> bsClose $ ldbBackingStore env
```
[the remark this answers is not in the document]
Comment (jasagredo) [reply ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/V1/Init.hs:309]: `closeDB` only exists in the `InitDB` datatype, it is not part of the LedgerDB api

File: ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/LedgerDB/StateMachine.hs (hunk 2 above)
Comment (jasagredo) [reply ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/LedgerDB/StateMachine.hs:463]: Right, but then I have to deal with them in every use site, no?

**Comment by jasagredo** [comment]
### Merge activity

* **Mar 22, 11:15 AM EDT**: @jasagredo started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/IntersectMBO/ouroboros-consensus/1005).

**Pull request merged**